src/include/platform/ICDEventManager.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+/* this file behaves like a config.h, comes first */
+#include "lib/core/CHIPError.h"
+#include <platform/ICDManager.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+namespace chip {
+namespace DeviceLayer {
+
+/**
+ * @brief ICDEventManager class is responsible of processing Platform Events that affect an ICDs behaviour
+ * The class registers an Event Handler with the Platform Manager and dispatchs the processing to the ICDManager class.
+ */
+class ICDEventManager
```

Comment:
if we start adding this, is there any way to template unit testing as well? We have a hard time unit testing things if we do not design them as such to begin with. I am unsure which parts of ICD generally can be unit tested.

Reply:
The goal is to have the event processing in the ICDEventManager class and the business logic in the ICDManager class.
Since the ICDManager will not be dependent of the EventLoop, it can and will be unit tested.

As for template unit testing, not sure what you mean.

Reply:
done